subsys/net/ip/net_pkt.c

```diff
@@ -396,18 +388,16 @@ struct net_buf *net_pkt_get_reserve_data(struct net_buf_pool *pool,
 		return NULL;
 	}
 
-	net_buf_reserve(frag, reserve_head);
-
 #if NET_LOG_LEVEL >= LOG_LEVEL_DBG
 	NET_FRAG_CHECK_IF_NOT_IN_USE(frag, frag->ref + 1);
 #endif
 
 	net_pkt_alloc_add(frag, false, caller, line);
 
-#if CONFIG_NET_PKT_LOG_LEVEL >= LOG_LEVEL_DBG
-	NET_DBG("%s (%s) [%d] frag %p reserve %u ref %d (%s():%d)",
+#if NET_LOG_LEVEL >= LOG_LEVEL_DBG
```

Comment:
We need to use CONFIG_NET_PKT_LOG_LEVEL here so that the debug prints are only printed if net_pkt debugging is enabled.  The CONFIG_NET_DEBUG_NET_PKT_ALLOC option depends on/sets NET_LOG_LEVEL so that we get function name/line number info into the alloc/free functions. But we do not want these extra alloc debug prints in that case as they are just printing too much info.